src/Server/TCPHandler.cpp

```diff
@@ -1110,8 +1111,13 @@ void TCPHandler::processInsertQuery(QueryState & state)
                 startInsertQuery(state);
 
             while (receivePacketsExpectDataConcurrentWithExecutor(state))
+            {
                 executor.push(std::move(state.block_for_insert));
 
+                sendLogs(state);
```

Comment:
Should we be sending progress too? It'd be nice to leave a comment explaining why we send some values and not others

Reply:
To be honest -- 'I do not know'
It was and it is still overcomplicated question what and when we send something.
I has checked the previous version of that code: 
progress has not been sent for each block when `processInsertQuery`

Lest try to add `sendProgress` and check the tests. It makes sence.

Reply:
No. If I add `sendProgress` tests fail.